packages/core/lib/utils.js

```diff
@@ -136,33 +138,46 @@ var utils = {
 
       if (xAmznTraceId) {
         var data = utils.processTraceData(xAmznTraceId);
-        valid = !!(data && data.Root && data.Parent && data.Sampled);
+        valid = !!(data && data.root && data.parent && data.sampled);
       }
 
       return valid;
     },
 
+    /**
+     * Populates trace ID, parent ID, and sampled decision of given segment. Will always populate valid values,
+     * even if xAmznTraceId contains missing or invalid values. This ensures downstream services receive valid
+     * headers.
+     * @param {Segment} segment - Facade segment to be populated
+     * @param {String} xAmznTraceId - Raw Trace Header to supply trace data
+     * @returns {Boolean} - true if required fields are present and Trace ID is valid, false otherwise
+     */
     populateTraceData: function(segment, xAmznTraceId) {
       logger.getLogger().debug('Lambda trace data found: ' + xAmznTraceId);
       var data = utils.processTraceData(xAmznTraceId);
-      var populated = false;
-
-      if (data && data.Root && data.Parent && data.Sampled) {
-        segment.trace_id = data.Root;
-        segment.id = data.Parent;
-
-        if (!parseInt(data.Sampled))
-          segment.notTraced = true;
-        else
-          delete segment.notTraced;
+      var valid = false;
+      
+      if (!data) {
+        data = {};
+        logger.getLogger().error('_X_AMZN_TRACE_ID is empty or has an invalid format');
+      } else if (!data.root || data.root !== TraceID.FromString(data.root).toString() 
+                 || !data.parent || !data.sampled) 
+      {
+        logger.getLogger().error('_X_AMZN_TRACE_ID is missing required information');
+      } else {
+        valid = true;
+      }
 
-        logger.getLogger().debug('Segment started: ' + JSON.stringify(data));
+      segment.trace_id = TraceID.FromString(data.root).toString();  // Will always assign valid trace_id
+      segment.id = data.parent || crypto.randomBytes(8).toString('hex');
 
-        populated = true;
-      } else
-        logger.getLogger().warn('_X_AMZN_TRACE_ID is missing required data.');
+      if (!parseInt(data.sampled))
```

Comment:
It looks like `parseInt` returns a 0 if the input is "?" which is a valid sampling decision ("Requested"). https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt

Can we create a SamplingDecision enum like Java?

Reply:
Yes, currently `?` is handled the same way as a 0 sampling decision because currently in the node SDK the sampling decision is modeled as strictly a boolean. While I agree it should be an enum eventually, changing the segment model to use an `enum` instead of a `boolean` would be a pretty big undertaking for this PR. What do you think about the priority?

Reply:
Yeah it doesn't have to be in this PR but I'm afraid it becomes another time bomb like the "dummy" trace id if we lost track of it. There's a big disagreement:`?` means "to be decided" while 0 means "not sampled". Shall we create an issue on fixing this?

Reply:
Totally agree. I've made #291 to track it.